lib/presentation/ui/home/home_page.dart

```diff
@@ -212,50 +230,52 @@ class _RecentListWidgetState extends State<RecentListWidget> {
 
   @override
   Widget build(BuildContext context) {
-    return StreamBuilder<List<HomeListItem>>(
-        stream: widget.bloc.occurrencesStream,
-        builder: (context, snapshot) {
-          if (snapshot.connectionState == ConnectionState.waiting) {
-            return Center(child: Text("A carregar"));
-          }
-
-          _refreshController.refreshCompleted();
-          if (snapshot.hasData) {
-            return Container(
-                color: Colors.white,
-                child: SmartRefresher(
-                  controller: _refreshController,
-                  header:  WaterDropMaterialHeader(backgroundColor: Theme.of(context).accentColor,),
-                  onRefresh: _onRefresh,
-                  enablePullDown: true,
-                  child: ListView.separated(
-                    separatorBuilder: (context, index) => Divider(
-                      indent: 50.0,
-                      thickness: 2.0,
+    return SafeArea(
+      child: StreamBuilder<List<HomeListItem>>(
+          stream: widget.isRecent ? widget.bloc.occurrencesStream : widget.bloc.favoritedOccurrencesStream,
+          builder: (context, snapshot) {
+            if (snapshot.connectionState == ConnectionState.waiting) {
+              return Center(child: Text("A carregar"));
+            }
+
+            _refreshController.refreshCompleted();
```

Comment:
Doesn't this trigger a new `setState` and thus need a post frame callback or move to somewhere else?

Reply:
Apparently, no

```
/// request complete,the header will enter complete state,
  ///
  /// resetFooterState : it will set the footer state from noData to idle
  void refreshCompleted({bool resetFooterState: false}) {
    headerMode?.value = RefreshStatus.completed;
    if (resetFooterState) {
      resetNoData();
    }
  }
```

I have also used this in my private projects and there seems to be no problem with it